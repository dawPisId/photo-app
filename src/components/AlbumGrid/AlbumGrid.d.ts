interface AlbumGridProps {
  children: JSX.Element;
}

declare const Grid: ({ children }: AlbumGridProps) => JSX.Element;
