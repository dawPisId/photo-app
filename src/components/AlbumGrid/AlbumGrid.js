import { Layout } from "./AlbumGrid.styled";
import React from "react";

const Grid = ({ children }) => <Layout>{children}</Layout>;

export default Grid;
