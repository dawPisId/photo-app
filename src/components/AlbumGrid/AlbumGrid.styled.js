import styled from "styled-components";

export const Layout = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 1rem;
`;
