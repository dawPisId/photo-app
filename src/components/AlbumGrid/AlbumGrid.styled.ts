import styled from "styled-components";

export const Layout = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 1rem;
  margin-bottom: 2rem;
  min-width: 0;
  min-height: 0;
  @media (max-width: 740px) {
    grid-template-columns: 1fr 1fr;
  }
  @media (max-width: 510px) {
    grid-template-columns: 1fr;
  }
`;
