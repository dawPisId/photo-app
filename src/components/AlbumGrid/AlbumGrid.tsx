import React, { FC } from "react";

import { Layout } from "./AlbumGrid.styled";

const AlbumGrid: FC<any> = ({ children }) => <Layout>{children}</Layout>;

export default AlbumGrid;
