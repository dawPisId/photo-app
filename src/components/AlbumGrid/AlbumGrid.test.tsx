import AlbumGrid from "./AlbumGrid";
import React from "react";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <AlbumGrid>
        <div>One</div>
        <div>Two</div>
        <div>Three</div>
        <div>Four</div>
      </AlbumGrid>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
