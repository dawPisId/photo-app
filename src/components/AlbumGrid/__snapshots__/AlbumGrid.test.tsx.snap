// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdnxRM iznVzs"
>
  <div>
    One
  </div>
  <div>
    Two
  </div>
  <div>
    Three
  </div>
  <div>
    Four
  </div>
</div>
`;
